TreeView Virtual Mode: rewrite (#988)
DevExpress/devextreme-demos: JSDemos/Demos/TreeView/VirtualMode/description.md

> With virtual mode, the TreeView loads a set of child nodes when their parent node is expanded. This enhances performance on large datasets.

Reviewer: ```suggestion
With virtual mode, the TreeView loads a set of child nodes only when their parent node is expanded. This enhances performance on large datasets.
```

> With virtual mode, the TreeView loads a set of child nodes only when their parent node is expanded. This enhances performance on large datasets.

Reviewer 2: ```suggestion
With virtual mode, the TreeView loads a node’s children when the node is expanded for the first time. This enhances performance on large datasets.
```

> With virtual mode, the TreeView loads a node’s children when the node is expanded for the first time. This enhances performance on large datasets.

Reviewer 3: ```suggestion
In virtual mode, the TreeView loads a node’s children when the node is expanded for the first time. This enhances performance on large datasets.
```

> To enable this feature, set the [virtualModeEnabled](/Documentation/ApiReference/UI_Components/dxTreeView/Configuration/#virtualModeEnabled) property to **true**. Note that the mode is only available when the [dataStructure](/Documentation/ApiReference/UI_Components/dxTreeView/Configuration/#dataStructure) is plain.

Reviewer 2: ```suggestion
To enable this feature, set the [virtualModeEnabled](/Documentation/ApiReference/UI_Components/dxTreeView/Configuration/#virtualModeEnabled) property to **true**. Note that this mode is only available when the TreeView's [dataStructure](/Documentation/ApiReference/UI_Components/dxTreeView/Configuration/#dataStructure) is plain.
```

> To enable this feature, set the [virtualModeEnabled](/Documentation/ApiReference/UI_Components/dxTreeView/Configuration/#virtualModeEnabled) property to **true**. Note that the mode is only available when the dataset's [structure](/Documentation/ApiReference/UI_Components/dxTreeView/Configuration/#dataStructure) is plain.

Reviewer: ```suggestion
To enable this feature, set the [virtualModeEnabled](/Documentation/ApiReference/UI_Components/dxTreeView/Configuration/#virtualModeEnabled) property to **true**. Note that the mode is only available when the [dataStructure](/Documentation/ApiReference/UI_Components/dxTreeView/Configuration/#dataStructure) is plain.
```

> When the data source is remote, the TreeView requests data even for those nodes that do not nest others. To prevent this as in this demo, set the [hasItems](/Documentation/ApiReference/UI_Components/dxTreeView/Configuration/items/#hasItems) field to **false** for the corresponding data objects. This example uses the [hasItemsExpr](/Documentation/ApiReference/UI_Components/dxTreeView/Configuration/#hasItemsExpr) property to specify a custom name for this field.

Reviewer 2: ```suggestion
When the data source is remote, the TreeView requests data even for those nodes that do not nest others. To prevent this, set the [hasItems](/Documentation/ApiReference/UI_Components/dxTreeView/Configuration/items/#hasItems) field to **false** for the corresponding data objects. This example also uses the [hasItemsExpr](/Documentation/ApiReference/UI_Components/dxTreeView/Configuration/#hasItemsExpr) property to specify a custom name for this field.
```

> When the data source is remote, the TreeView requests data even for those nodes that do not nest others. To prevent this, set the [hasItems](/Documentation/ApiReference/UI_Components/dxTreeView/Configuration/items/#hasItems) field to **false** for the corresponding data objects. This example also uses the [hasItemsExpr](/Documentation/ApiReference/UI_Components/dxTreeView/Configuration/#hasItemsExpr) property to specify a custom name for this field.

Reviewer 3: ```suggestion
When you use a remote data source, the TreeView also requests data for nodes that contain nested nodes. To prevent this, set the [hasItems](/Documentation/ApiReference/UI_Components/dxTreeView/Configuration/items/#hasItems) field to **false** for the corresponding data objects. This example also uses the [hasItemsExpr](/Documentation/ApiReference/UI_Components/dxTreeView/Configuration/#hasItemsExpr) property to specify a custom name for this field.
```

> As an alternative, you can use the [createChildren](/Documentation/ApiReference/UI_Components/dxTreeView/Configuration/#createChildren) function to gain full control over data load as done in the [Load Data on Demand](/Demos/WidgetsGallery/Demo/TreeView/LoadDataOnDemand) demo.

Reviewer 4: I would add 'as an alternative to virtual mode' or something like that to make it clear what is createChildren alternative to.

> As an alternative to virtual mode, you can use the [createChildren](/Documentation/ApiReference/UI_Components/dxTreeView/Configuration/#createChildren) function to gain full control over data load as done in the [Load Data on Demand](/Demos/WidgetsGallery/Demo/TreeView/LoadDataOnDemand) demo.

Reviewer 2: ```suggestion
As an alternative to virtual mode, you can use the [createChildren](/Documentation/ApiReference/UI_Components/dxTreeView/Configuration/#createChildren) function to gain full control over data load. To see this approach in action, refer to the [Load Data on Demand](/Demos/WidgetsGallery/Demo/TreeView/LoadDataOnDemand) demo.
```

> As an alternative to virtual mode, you can use the [createChildren](/Documentation/ApiReference/UI_Components/dxTreeView/Configuration/#createChildren) function to gain full control over data load. To see this approach in action, refer to the [Load Data on Demand](/Demos/WidgetsGallery/Demo/TreeView/LoadDataOnDemand) demo.

Reviewer 3: ` to gain full control` means what?

Author: To specify a function that processes the requested data in a custom way. Can the sentence be
```suggestion
As an alternative to virtual mode, you can use the [createChildren](/Documentation/ApiReference/UI_Components/dxTreeView/Configuration/#createChildren) function to process requested data in a custom way. To see this approach in action, refer to the [Load Data on Demand](/Demos/WidgetsGallery/Demo/TreeView/LoadDataOnDemand) demo.
```
?

Reviewer 3: 'in a custom way' does not explain anything nor should you consider it as the correct way to express something in a technical context

Author: "As an alternative to virtual mode, you can use a custom logic to process requested data. To do this, specify the createChildren function as shown in the Load Data on Demand demo." - what about this?

Reviewer 3: Ok

> In virtual mode, the TreeView loads a node’s children when the node is expanded for the first time. This enhances performance on large datasets.

Reviewer: ```suggestion
In virtual mode, the TreeView loads a node's children when the node is expanded for the first time. This enhances performance on large datasets.
```
